admin/class-yoast-notification-center.php

```diff
@@ -538,10 +561,21 @@ public function update_storage() {
 			return;
 		}
 
-		$notifications = array_map( array( $this, 'notification_to_array' ), $notifications );
+		$this->store_notifications( $notifications );
+	}
 
-		// Save the notifications to the storage.
-		update_user_option( get_current_user_id(), self::STORAGE_KEY, $notifications );
+	/**
+	 * Stores the notifications to its respective user's storage.
+	 *
+	 * @param array|Yoast_Notification[] $notifications The notifications to store.
+	 *
+	 * @return void
+	 */
+	private function store_notifications( $notifications ) {
```

Comment:
With my suggestion to do `array_walk`for storing the notification for each user you can change this to have:
```suggestion
	private function store_notifications( $notifications_per_user, $user_id ) {
```

By doing this you can remove the foreach and just call:
```
			$notifications_per_user = array_map( array( $this, 'notification_to_array' ), $notifications_per_user );
			update_user_option( $user_id, self::STORAGE_KEY, $notifications_per_user );
```